### ISSUE TITLE
Remotec zxt-120 A/C Controller

### ISSUE THREAD

**Issue by Auto-m8r** [ilcato/homebridge-Fibaro-HC2#55]
Hi Ilcato,

I love the work you have done with the Fibaro HC2 home bridge plugin. 

I have found that my Remotec ZXT-120 A/C Controller does not work in homebridge, it did detect before i updated to 1.09 however it still did not work correctly.

Is there a way to get this device working correctly?

**Comment by ilcato** [comment]
Did it work before? Really? I never supported it.

**Comment by Auto-m8r** [comment]
Homebridge imported it and it displayed in the home app. The device didn't really work though as most of the functions were not working. If i adjusted the temp on my iPhone the A/C unit would power on however i couldn't control heat, cool or turn off after that.

it would be great if it did work though.

**Comment by Mx-Razvan** [comment]
Hi!
now I find my answer to the same problem. :(
In fact my remotec worked quite well after selecting all in HCL - remotec - Parameter 35
Learn location status and save.
Indeed - would be great to have it working again

**Comment by ilcato** [comment]
@Auto-m8r , @Mx-Razvan,
can you send me the result of the following query:

`http://YOUR_HC_IP_ADDRESS:80/api/devices/DEVICE_ID`

where YOUR_HC_IP_ADDRESS is the address of your Home Center and DEVICE_ID is the ID of the Remotec device.

**Comment by Mx-Razvan** [comment]
Will do asap.

Thanks,
R

**Comment by Mx-Razvan** [comment]
but where I have to submit the above command?

**Comment by ilcato** [comment]
In a browser from a computer connected to the same network as the Home Center

**Comment by Mx-Razvan** [comment]
sorry - already tried but by mistake I typed 88 instead 80.
here we go:

this is the main device (it has 4 branches):

{"id":61,"name":"Remotec","roomID":1,"type":"com.fibaro.zwaveDevice","baseType":"com.fibaro.device","enabled":true,"visible":false,"isPlugin":false,"parentId":1,"remoteGatewayId":0,"interfaces":["battery","zwave","zwaveConfiguration","zwaveSlaveRouting"],"properties":{"batteryLevel":58,"batteryLowNotification":"true","configured":"true","dead":"false","defInterval":"0","deviceControlType":"0","deviceIcon":"28","deviceSpecificData":"","deviceSpecificIdType":"","emailNotificationID":"0","emailNotificationType":"0","endPointId":"0","liliOffCommand":"","liliOnCommand":"","log":"","logTemp":"","manufacturer":"","markAsDead":"true","maxInterval":"0","minInterval":"0","model":"","nodeId":"16","parameters":[{"id":25,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":26,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":27,"lastReportedValue":429,"lastSetValue":429,"size":2,"value":429},{"id":28,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":32,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":33,"lastReportedValue":1,"lastSetValue":1,"size":1,"value":1},{"id":35,"lastReportedValue":0,"lastSetValue":0,"size":4,"value":0},{"id":37,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0}],"parametersTemplate":"417","pollingTimeSec":0,"productInfo":"82,84,1,1,131,119,1,20","pushNotificationID":"0","pushNotificationType":"0","remoteGatewayId":"0","saveLogs":"true","serialNumber":"","smsNotificationID":"0","smsNotificationType":"0","stepInterval":"0","useTemplate":"true","userDescription":"","zwaveCompany":"Remotec Technology Ltd","zwaveInfo":"6,3,42","zwaveSoftwareVersion":"{}","zwaveVersion":"1.20"},"actions":{"getParameter":1,"reconfigure":0,"setInterval":1,"setParameter":2},"created":1491189604,"modified":1491189604,"sortOrder":55} 

and this is the ac control where I set the temperature, cool/heat/dry so on. this used to work in homekit:

{"id":62,"name":"AC Control","roomID":1,"type":"com.fibaro.setPoint","baseType":"com.fibaro.hvac","enabled":true,"visible":true,"isPlugin":false,"parentId":61,"remoteGatewayId":0,"interfaces":["battery","zwave","zwaveSwitchAll"],"properties":{"batteryLevel":58,"parameters":[{"id":25,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":26,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":27,"lastReportedValue":429,"lastSetValue":429,"size":2,"value":429},{"id":28,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":32,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":33,"lastReportedValue":1,"lastSetValue":1,"size":1,"value":1},{"id":35,"lastReportedValue":0,"lastSetValue":0,"size":4,"value":0},{"id":37,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0}],"pollingTimeSec":0,"zwaveCompany":"Remotec Technology Ltd","zwaveInfo":"6,3,42","zwaveVersion":"1.20","batteryLowNotification":"true","configured":"true","dead":"false","defInterval":"0","deviceControlType":"0","deviceIcon":"71","emailNotificationID":"0","emailNotificationType":"0","endPointId":"0","liliOffCommand":"","liliOnCommand":"","log":"","logTemp":"","manufacturer":"","markAsDead":"true","maxInterval":"0","minInterval":"0","mode":"1","model":"","nodeId":"16","parametersTemplate":"417","productInfo":"82,84,1,1,131,119,1,20","pushNotificationID":"0","pushNotificationType":"0","remoteGatewayId":"0","saveLogs":"true","serialNumber":"","smsNotificationID":"0","smsNotificationType":"0","stepInterval":"0","supportedModes":"1,2,8","switchAllMode":"SWITCH_ALL_EXCLUDED_FROM_THE_ALL_ON_ALL_OFF_FUNCTIONALITY","targetLevel":"28.00","unit":"C","useTemplate":"true","userDescription":"","value":"28.00"},"actions":{"reconfigure":0,"setInterval":1,"setSetpointMode":1,"setThermostatSetpoint":2},"created":1491189604,"modified":1491189604,"sortOrder":47}

also there are two more branches - a temperature sensor inside:

{"id":63,"name":"Temperatura SENZOR","roomID":1,"type":"com.fibaro.temperatureSensor","baseType":"com.fibaro.multilevelSensor","enabled":true,"visible":true,"isPlugin":false,"parentId":61,"remoteGatewayId":0,"interfaces":["battery","zwave"],"properties":{"batteryLevel":58,"parameters":[{"id":25,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":26,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":27,"lastReportedValue":429,"lastSetValue":429,"size":2,"value":429},{"id":28,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":32,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":33,"lastReportedValue":1,"lastSetValue":1,"size":1,"value":1},{"id":35,"lastReportedValue":0,"lastSetValue":0,"size":4,"value":0},{"id":37,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0}],"pollingTimeSec":0,"zwaveCompany":"Remotec Technology Ltd","zwaveInfo":"6,3,42","zwaveVersion":"1.20","batteryLowNotification":"true","configured":"true","dead":"false","defInterval":"0","deviceControlType":"0","deviceIcon":"30","emailNotificationID":"0","emailNotificationType":"0","endPointId":"0","liliOffCommand":"","liliOnCommand":"","log":"","logTemp":"","manufacturer":"","markAsDead":"true","maxInterval":"0","minInterval":"0","model":"","nodeId":"16","offset":"0.00","parametersTemplate":"417","productInfo":"82,84,1,1,131,119,1,20","pushNotificationID":"0","pushNotificationType":"0","remoteGatewayId":"0","saveLogs":"true","serialNumber":"","showFireAlarm":"true","showFreezeAlarm":"false","smsNotificationID":"0","smsNotificationType":"0","stepInterval":"0","unit":"C","useTemplate":"true","userDescription":"","value":"26.00"},"actions":{"reconfigure":0,"setInterval":1},"created":1491189604,"modified":1491189604,"sortOrder":49}

a operation mode branch - cool/heat/ dry/resume:

{"id":64,"name":"Mod OP AC","roomID":1,"type":"com.fibaro.operatingMode","baseType":"com.fibaro.hvac","enabled":true,"visible":true,"isPlugin":false,"parentId":61,"remoteGatewayId":0,"interfaces":["battery","zwave"],"properties":{"batteryLevel":58,"parameters":[{"id":25,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":26,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":27,"lastReportedValue":429,"lastSetValue":429,"size":2,"value":429},{"id":28,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":32,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":33,"lastReportedValue":1,"lastSetValue":1,"size":1,"value":1},{"id":35,"lastReportedValue":0,"lastSetValue":0,"size":4,"value":0},{"id":37,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0}],"pollingTimeSec":0,"zwaveCompany":"Remotec Technology Ltd","zwaveInfo":"6,3,42","zwaveVersion":"1.20","batteryLowNotification":"true","configured":"true","dead":"false","defInterval":"0","deviceControlType":"0","deviceIcon":"70","emailNotificationID":"0","emailNotificationType":"0","endPointId":"0","liliOffCommand":"","liliOnCommand":"","log":"","logTemp":"","manufacturer":"","markAsDead":"true","maxInterval":"0","minInterval":"0","mode":"0","model":"","nodeId":"16","parametersTemplate":"417","productInfo":"82,84,1,1,131,119,1,20","pushNotificationID":"0","pushNotificationType":"0","remoteGatewayId":"0","saveLogs":"true","serialNumber":"","smsNotificationID":"0","smsNotificationType":"0","state":"0","stepInterval":"0","supportedModes":"0,1,2,3,5,6,8","unit":"","useTemplate":"true","userDescription":"","value":"0.00"},"actions":{"reconfigure":0,"setInterval":1,"setMode":1},"created":1491189604,"modified":1491189604,"sortOrder":48}

and a fan branch - low speed, high speed...:

{"id":65,"name":"Ventilator AC","roomID":1,"type":"com.fibaro.fanMode","baseType":"com.fibaro.hvac","enabled":true,"visible":true,"isPlugin":false,"parentId":61,"remoteGatewayId":0,"interfaces":["battery","zwave"],"properties":{"batteryLevel":58,"parameters":[{"id":25,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":26,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0},{"id":27,"lastReportedValue":429,"lastSetValue":429,"size":2,"value":429},{"id":28,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":32,"lastReportedValue":255,"lastSetValue":255,"size":1,"value":255},{"id":33,"lastReportedValue":1,"lastSetValue":1,"size":1,"value":1},{"id":35,"lastReportedValue":0,"lastSetValue":0,"size":4,"value":0},{"id":37,"lastReportedValue":0,"lastSetValue":0,"size":1,"value":0}],"pollingTimeSec":0,"zwaveCompany":"Remotec Technology Ltd","zwaveInfo":"6,3,42","zwaveVersion":"1.20","batteryLowNotification":"true","configured":"true","dead":"false","defInterval":"0","deviceControlType":"0","deviceIcon":"73","emailNotificationID":"0","emailNotificationType":"0","endPointId":"0","liliOffCommand":"","liliOnCommand":"","log":"","logTemp":"","manufacturer":"","markAsDead":"true","maxInterval":"0","minInterval":"0","mode":"4","model":"","nodeId":"16","parametersTemplate":"417","productInfo":"82,84,1,1,131,119,1,20","pushNotificationID":"0","pushNotificationType":"0","remoteGatewayId":"0","saveLogs":"true","serialNumber":"","smsNotificationID":"0","smsNotificationType":"0","state":"0","stepInterval":"0","supportedModes":"0,1,2,3,4,5,128","unit":"","useTemplate":"true","userDescription":"","value":"0.00"},"actions":{"reconfigure":0,"setFanMode":1,"setInterval":1},"created":1491189604,"modified":1491189604,"sortOrder":46}

**Comment by ilcato** [comment]
Understood. I removed support for com.fibaro.setPoint.
How wa the user interface for "AC Control" in the home app?

**Comment by Auto-m8r** [comment]
@ilcato

I can see @Mx-Razvan already posted the results, do you require mine also?

As for your question above, the multiple Fibaro devices presented as one device in the home app. It was quite good however as I originally posted mostly unusable for me.

**Comment by Mx-Razvan** [comment]
Yes - multiple fibaro devices presented as one in home app, with the AC control functions displayed. When I gave commands in home app the modifications were just in this branch in fibaro system

**Comment by Auto-m8r** [comment]
@Mx-Razvan

What is parameter 35 used for and what do you have it set to?

If @ilcato can get this functioning again for us I will need to set this so mine works correctly?

**Comment by Mx-Razvan** [comment]
As Remotec is not very generous with this parameter role I think that the AC controller activate the IR code map learned before from the AC remote control. 
In my case after the first homebridge and fibaro plugin install the homekit showed AC control switch in the home app. It has switches for cool / heat / dry and a slider for temperature but it did nothing. After checking all items in this parameter list and save - it started to function normally and an OFF option was added to the home app switch. And all the functions, including the temperature setup were ok and working.
After a forced reinstall of homebridge and fibaro plugin (1.0.9) at the fibaro accessories import in homekit the AC control accessory is no more imported and showed.

**Comment by ilcato** [comment]
@Mx-Razvan this is very strange. It is true that there was a support for com.fibaro.setPoint devices in terms of visualisation of an HomeKit accessory but there was no support for controlling state. There was a support for setting and reading temperature but nothing all. I will try to support it in the next week end.

**Comment by Auto-m8r** [comment]
Thanks @ilcato I appreciate your hard work on this. I'm happy to assist with whatever testing you may require.

**Comment by Mx-Razvan** [comment]
@ilcato yes - it is. After the initial install of homebridge , fibaro plugin and accessories import I started to see the on the home app the entrance (the physical location of remotec) temperature - 30 degrees and two more temperature (from other sensors, apart from remotec, less than 30 degrees and very close as values). And I said what the hell is this - can't be so warm there - at the remotec level, so I started to play with it and saw that actually home app read the last temperature set from fibaro in "ac control" and not from the temperature sensor embedded in the same device. Trying to control the device at that stage - no reaction. Then I played a little in Fibaro as I said and the result was that it started to work normally.
Thanks for your work and looking forward for support. Let me know if you need anything to be tested or additional details are needed.

**Comment by ilcato** [comment]
@Mx-Razvan , @Auto-m8r ,
please try the just committed version of the index.js file: replace the existing version in your setup and restart homebridge.

**Comment by Mx-Razvan** [comment]
@ilcato 
Jus did this. No new item in the accesory list.

**Comment by Mx-Razvan** [comment]
@ilcato 
Now it appears 
![img_0014](https://cloud.githubusercontent.com/assets/26866849/24839410/6a85d8e2-1d62-11e7-9066-2f25298e7ebf.PNG)
![img_0015](https://cloud.githubusercontent.com/assets/26866849/24839411/6ab2cd0c-1d62-11e7-9e4a-7f761d7ccdbe.PNG)

Like last time first is for cool, heat, auto and off and the other slider is for temperature setup. Do not see the "dry" option but isn't necessary

**Comment by Mx-Razvan** [comment]
After some playing with it seems that now the temperature setup is working fine in the AC unit predefined value (16 - 30 degrees). Unfortunatelly now the on/off cool, heat buttons aren't working. Originaly in fibaro the operation mode is set by device no 64 from above - mod op ac:
![img_0018](https://cloud.githubusercontent.com/assets/26866849/24839719/9f4636d4-1d68-11e7-9f25-a53697258af7.PNG)
If the fibaro is accesed through a browser. The AC controls:
![img_0017](https://cloud.githubusercontent.com/assets/26866849/24839736/c99e0d58-1d68-11e7-8e9d-bead365b060e.PNG) aren't working. 
So a working AC control should have mixed functions from com.fibaro.setPoint and com.fibaro.operatingMode.
The strange thing is that in fibaro ipad app the AC Control (62 from above) is working and it sends commands for heat/cool/dry and they are transmitted to the AC unit:
![img_0020](https://cloud.githubusercontent.com/assets/26866849/24839795/b232b686-1d69-11e7-9d76-dd3841e7bf35.PNG) but it has no OFF option.

**Comment by Auto-m8r** [comment]
I to have tested this and can confirm that I now have the device visible in HomeKit however I can only control temperature as was the case before.

**Comment by ilcato** [comment]
@Mx-Razvan,
can you play with the api interface via http://HC2_IP/docs ?
You should go the devices section and select /devices/{deviceID}/action/{actionName}

put:

- 62 in deviceID
- mode in actionName
- {
  "args": [
    1
  ]
} in body


try different values for the number in [] and try to figure out the correspondence between the numbers and the modes.

Unfortunately I do not have this device and cannot test it.

**Comment by Auto-m8r** [comment]
Hi @ilcato 

I wanted to assist in testing this however after reading through the instruction above and @Mx-Razvan's previous posts I am confused as to which device we should be testing this against?

My setup is similar to @Mx-Razvan , with the below used devices:

Device 27
<img width="219" alt="screen shot 2017-04-17 at 9 35 19 pm" src="https://cloud.githubusercontent.com/assets/26859568/25087915/d094ab46-23b5-11e7-9275-a27502fd0d20.png">

and Device 29
<img width="226" alt="screen shot 2017-04-17 at 9 36 28 pm" src="https://cloud.githubusercontent.com/assets/26859568/25088020/6ec1bc0a-23b6-11e7-8458-7a75ebee0f88.png">

I use device 29 to turn the AC on by selecting either "cooling" or "heating" mode and then I turn it off by selecting "off" mode, device 27 is used to set the desired temperature only.

regardless I tried running your commands against both devices and nothing happened at all so I suspect I am using the wrong commands perhaps? below is a screenshot off the window:
![img_0672 png-1](https://cloud.githubusercontent.com/assets/26859568/25088089/00096aaa-23b7-11e7-81e1-530b733a8d27.png)

What am I doing wrong?

**Comment by ilcato** [comment]
Did you tried values 1, 2, 3, 4, 5 beside the 10 I see?

**Comment by ilcato** [comment]
Try also setMode or setSetpointMode instead of mode

**Comment by Auto-m8r** [comment]
Thanks for the tip @il, setMode is the one we want. I'll run through some testing tomorrow for the various modes as it's a little late in the evening here now.

Thanks

**Comment by Auto-m8r** [comment]
@ilcato,

I have confirmed the following using setMode on device 29.

0 = off
1 = heat
2 = cool
5 = resume (whatever last state was before powering off)
8 = Dry Air

let me know if you need me to test further.

FYI Device 29 details:

"id": 29,
  "name": "Status AC",
  "roomID": 7,
  "type": "com.fibaro.operatingMode",
  "baseType": "com.fibaro.hvac",
  "enabled": true,
  "visible": true,
  "isPlugin": false,
  "parentId": 26,
  "remoteGatewayId": 0,
  "interfaces": [
    "battery",
    "zwave"
  ],
  "properties": {
    "batteryLevel": 100,
    "parameters": [
      {
        "id": 27,
        "lastReportedValue": 0,
        "lastSetValue": 0,
        "size": 2,
        "value": 0
      },
      {
        "id": 25,
        "lastReportedValue": 22,
        "lastSetValue": 22,
        "size": 1,
        "value": 22
      }
    ],
    "pollingTimeSec": 0,
    "zwaveCompany": "Remotec",
    "zwaveInfo": "6,3,42",
    "zwaveVersion": "1.20",
    "batteryLowNotification": "true",
    "configured": "true",
    "dead": "false",
    "defInterval": "0",
    "deviceControlType": "0",
    "deviceIcon": "70",
    "emailNotificationID": "0",
    "emailNotificationType": "0",
    "endPointId": "0",
    "liliOffCommand": "",
    "liliOnCommand": "",
    "log": "",
    "logTemp": "",
    "manufacturer": "",
    "markAsDead": "true",
    "maxInterval": "0",
    "minInterval": "0",
    "mode": "0",
    "model": "",
    "nodeId": "9",
    "parametersTemplate": "0",
    "productInfo": "82,84,1,2,131,119,1,20",
    "pushNotificationID": "0",
    "pushNotificationType": "0",
    "remoteGatewayId": "0",
    "saveLogs": "true",
    "serialNumber": "",
    "smsNotificationID": "0",
    "smsNotificationType": "0",
    "state": "0",
    "stepInterval": "0",
    "supportedModes": "0,1,2,5,6,8,10",
    "unit": "",
    "useTemplate": "false",
    "userDescription": "",
    "value": "0.00"
  },
  "actions": {
    "reconfigure": 0,
    "setInterval": 1,
    "setMode": 1
  },
  "created": 1492420135,
  "modified": 1492420135,
  "sortOrder": 13
}

**Comment by Mx-Razvan** [comment]
@ilcato  Hi! I was away from my home for a while. Let me know what you need further.
@Auto-m8r - per my understanding your device 29 is the one with the operation mode control, right? Similar with my 64.

**Comment by Auto-m8r** [comment]
Correct @Mx-Razvan, My device 29 is equal to your Device 64

**Comment by ilcato** [comment]
Thinking about how to correlate from the Fibaro API the two device in order to create a single HomeKit accessory that can control both the mode and temperature.

**Comment by Auto-m8r** [comment]
That would be the ideal solution if it can be done

**Comment by Mx-Razvan** [comment]
Hi @ilcato & @Auto-m8r 

Browsing through fibaro interface I saw a new plugin - mock device - allowing to create virtual devices from all kind, including like above ac control and operation mode. Still haven't figured it out how to link new created devices with the real devices but maybe it could be an idea for a single device to be integrated, after its setup, in fibaro plugin

**Comment by ilcato** [comment]
which plugin @Mx-Razvan ?

**Comment by Auto-m8r** [comment]
I can’t seem to find it either @Mx-Razvan? 

I have however since created a virtual device that works in HomeBridge to turn the AC on and Off.

**Comment by Mx-Razvan** [comment]
@ilcato - thiss one:
![mock](https://cloud.githubusercontent.com/assets/26866849/25607742/a98a936e-2f20-11e7-8b35-bc2a8d4c3cd4.jpg)

**Comment by Auto-m8r** [comment]
Oddly this does not appear under plugins for me?

**Comment by Mx-Razvan** [comment]
@Auto-m8r - do you have 4.120?

Speaking about your VD for on/off - could you write it here?

**Comment by Mx-Razvan** [comment]
As I updated recently from HCL to HC-2 I noted that this plugin is no available in HC-2, only in HCL. 

Apart of this - any news on this subject?

**Comment by ilcato** [comment]
No news, I'involved in a complete rewriting of the plugin.
Regarding HC2 the plugin is perfectly compatible, I have an HC2 and developed the plugin for it. It also worked on HCL but as a secondary positive effect.

**Comment by agisofttm** [comment]
Hmmm, can someone help about Remotec issue.... We just added Remotec ZXT-120 to fibaro, restarted homebridge and i can't see it in Accessories...... I've stopped homebridge service, and started manualy and have this output....

[6/9/2017, 8:17:17 PM] [Dima Style] Error getting data from Home Center:  TypeError: Cannot read property 'getCharacteristic' of undefined
    at Accessory._sideloadServices (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/Accessory.js:880:5)
    at PlatformAccessory._prepareAssociatedHAPAccessory (/usr/lib/node_modules/homebridge/lib/platformAccessory.js:148:32)
    at Server._handleRegisterPlatformAccessories (/usr/lib/node_modules/homebridge/lib/server.js:431:15)
    at Server.<anonymous> (/usr/lib/node_modules/homebridge/lib/server.js:36:10)
    at emitOne (events.js:96:13)
    at API.emit (events.js:191:7)
    at API.registerPlatformAccessories (/usr/lib/node_modules/homebridge/lib/api.js:159:8)
    at ShadowAccessory.resgisterUpdateccessory (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/shadows.js:88:17)
    at FibaroHC2.addAccessory (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/index.js:146:25)
    at LoadAccessories.devices.map (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/index.js:102:22)

**Comment by ilcato** [comment]
@agisofttm, which version of the plugin do you have. Please upgrade to the last version: 2.0.3

**Comment by agisofttm** [comment]
@ilcato  we are with 2.0.1

pi@Raspberry:~ $ sudo npm ls -g | grep fibaro
├─┬ homebridge-fibaro-hc2@2.0.1
pi@Raspberry:~ $

**Comment by agisofttm** [comment]
Forgot to mention..... my fibaro is with new beta from yesterday 4.131 - if there is some difference of course

**Comment by ilcato** [comment]
@agisofttm Please upgrade to the last version: 2.0.3

**Comment by agisofttm** [comment]
@ilcato hmm... updated to 2.0.3, and now I have other new devices, but this one particular, and one energy meter are missing.... should I delete folders /accessories and /persist and try again?

**Comment by ilcato** [comment]
It should not be necessary. try to delete only accessories folder

**Comment by agisofttm** [comment]
@ilcato ... nope... still the same.... 

[6/9/2017, 9:39:55 PM] [Dima Style] Added/changed accessory:  Вентилация
[6/9/2017, 9:39:55 PM] [Dima Style] Added/changed accessory:  154.2
[6/9/2017, 9:39:55 PM] [Dima Style] Added/changed accessory:  Осв. между етажи
[6/9/2017, 9:39:55 PM] [Dima Style] Error getting data from Home Center:  TypeError: Cannot read property 'getCharacteristic' of undefined
    at Accessory._sideloadServices (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/Accessory.js:880:5)
    at PlatformAccessory._prepareAssociatedHAPAccessory (/usr/lib/node_modules/homebridge/lib/platformAccessory.js:148:32)
    at Server._handleRegisterPlatformAccessories (/usr/lib/node_modules/homebridge/lib/server.js:431:15)
    at Server.<anonymous> (/usr/lib/node_modules/homebridge/lib/server.js:36:10)
    at emitOne (events.js:96:13)
    at API.emit (events.js:191:7)
    at API.registerPlatformAccessories (/usr/lib/node_modules/homebridge/lib/api.js:159:8)
    at ShadowAccessory.resgisterUpdateccessory (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/shadows.js:88:17)
    at FibaroHC2.addAccessory (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/index.js:158:25)
    at LoadAccessories.devices.map (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/index.js:105:22)
[6/9/2017, 9:40:02 PM] Mute : [false]
[6/9/2017, 9:40:02 PM] Busy with alarm now - not getting status
[6/9/2017, 9:40:02 PM] Got status
[6/9/2017, 9:40:02 PM] Results:

any other ideas?

**Comment by ilcato** [comment]
Check to see if you are really on 2.0.3

**Comment by ilcato** [comment]
let me see your config.json

**Comment by agisofttm** [comment]
@ilcato - there it is

pi@Raspberry:~/.homebridge $ sudo npm list -g | grep fibaro
├─┬ homebridge-fibaro-hc2@2.0.3
pi@Raspberry:~/.homebridge $

and config.json

pi@Raspberry:~/.homebridge $ cat config.json
{
    "bridge": {
        "name": "Дима Стил",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154"
    },
    "platforms": [
        {
            "platform": "FibaroHC2",
            "name": "Dima Style",
            "host": "192.168.2.57",
            "username": "vladimirtodorov@dimastyle.com",
            "password": "****",
            "grouping": "room",
	    "pollerperiod": "2",
            "securitysystem": "disabled"
        },
	{
            "platform": "Paradox",
            "ip": "192.168.2.54",
            "port": 8084,
            "password": "********",
            "accessories": [
        		{
        		"name": "Alarm",
        		"config": {
        			"type": "Alarm"
        			}
        	 	},
        	 	
        		{
        		"name": "Офис Влади",
        		"config": {
        			"type": "Motion Sensor",
        			"zone": 20,
        			"pgm": 0
        			}
        	 	},
        	 	

        		{
        		"name": "Етаж 2 ВИП",
        		"config": {
        			"type": "Motion Sensor",
        			"zone": 19
        			}
        	 	},
        	 	
        	 	
        		{
        		"name": "Етаж 2 Ап.1",
        		"config": {
        			"type": "Contact Sensor",
        			"zone": 4
        			}
        	 	},
        	 	
        	 	
         		{
        		"name": "Етаж 2 Ап.2",
        		"config": {
        			"type": "Contact Sensor",
        			"zone": 22
        			}

        	 	}        	 	
        	 ]
         }
    ],
    "accessories": [
    ]
}
pi@Raspberry:~/.homebridge $

**Comment by ilcato** [comment]
Try to add the "switchglobalvariables" parameter in config.json with a value of "".

**Comment by agisofttm** [comment]
@ilcato - tried, but same error. Actually fibaro plugin is working, but those devices is not shown - it generates error and continiue to working to next plugin  ....

pi@Raspberry:~/.homebridge $ homebridge 
*** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs>
*** WARNING *** The program 'nodejs' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs&f=DNSServiceRegister>
[6/10/2017, 11:52:38 AM] Loaded plugin: homebridge-fibaro-hc2
[6/10/2017, 11:52:38 AM] Registering platform 'homebridge-fibaro-hc2.FibaroHC2'
[6/10/2017, 11:52:38 AM] ---
[6/10/2017, 11:52:38 AM] Loaded plugin: homebridge-paradox-securitysystem
[6/10/2017, 11:52:38 AM] Registering platform 'homebridge-paradox.Paradox'
[6/10/2017, 11:52:38 AM] ---
[6/10/2017, 11:52:38 AM] Loaded plugin: homebridge-server
[6/10/2017, 11:52:38 AM] Registering platform 'homebridge-server.Server'
[6/10/2017, 11:52:38 AM] ---
[6/10/2017, 11:52:38 AM] Loaded config.json with 0 accessories and 2 platforms.
[6/10/2017, 11:52:38 AM] ---
[6/10/2017, 11:52:39 AM] Loading 2 platforms...
[6/10/2017, 11:52:39 AM] [Dima Style] Initializing FibaroHC2 platform...
[6/10/2017, 11:52:39 AM] Initializing Paradox platform...
[6/10/2017, 11:52:39 AM] Looking for accessories in config file
[6/10/2017, 11:52:39 AM] Accessory
[6/10/2017, 11:52:39 AM] Alarm
[6/10/2017, 11:52:39 AM] Found: Alarm [Alarm]
[6/10/2017, 11:52:39 AM] Accessory
[6/10/2017, 11:52:39 AM] Офис Влади
[6/10/2017, 11:52:39 AM] Found: Офис Влади [Motion Sensor]
[6/10/2017, 11:52:39 AM] Accessory
[6/10/2017, 11:52:39 AM] Етаж 2 ВИП
[6/10/2017, 11:52:39 AM] Found: Етаж 2 ВИП [Motion Sensor]
[6/10/2017, 11:52:39 AM] Accessory
[6/10/2017, 11:52:39 AM] Етаж 2 Ап.1
[6/10/2017, 11:52:39 AM] Found: Етаж 2 Ап.1 [Contact Sensor]
[6/10/2017, 11:52:39 AM] Accessory
[6/10/2017, 11:52:39 AM] Етаж 2 Ап.2
[6/10/2017, 11:52:39 AM] Found: Етаж 2 Ап.2 [Contact Sensor]
[6/10/2017, 11:52:39 AM] Initializing platform accessory 'Alarm'...
[6/10/2017, 11:52:39 AM] Initializing platform accessory 'Офис Влади'...
[6/10/2017, 11:52:39 AM] Initializing platform accessory 'Етаж 2 ВИП'...
[6/10/2017, 11:52:39 AM] Initializing platform accessory 'Етаж 2 Ап.1'...
[6/10/2017, 11:52:39 AM] Initializing platform accessory 'Етаж 2 Ап.2'...
[6/10/2017, 11:52:39 AM] Loading 0 accessories...
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Диспенсер
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Ап.1 Движение
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Ап.1 Температура
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Ап. 1 Осветеност
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Ап.1 Бойлер Офиси
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Бойлер Офиси
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Площандка
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  да се ползва за Вент
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  UPS Стая Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  ВакуАктив
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Врата Стая Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Темп. Стая Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Кафе Машина
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Клима Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Офис Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  UPS Eaton
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Дим Фризьoри 
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Темп.Въздух
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Реклама
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Педикюр
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  48V Захранване
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Сървър Вентилация 
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Фризьори
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  ДНК
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Аудио
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Рецепция Движение
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Рецепция
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Осветеност
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Врата
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  led рецепция
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Влажност
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Външна Температура
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Влажност
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Подово Рецепция
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Темп. Рецепция
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Подово
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Наводнение
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Темп. Наводнение
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Трева Напояване
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Капково Напояване
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Движение
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Бойлер
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Сървър Шкаф
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Темп. Движение
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Пералня
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Сушилня
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  UPS APC SC1000
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Бойлер
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  2️⃣
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  3️⃣
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  1️⃣
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Вентилация
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  154.2
[6/10/2017, 11:52:39 AM] [Dima Style] Configured Accessory:  Осв. между етажи
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 031-45-154 │     
    └────────────┘     
                       
[6/10/2017, 11:52:39 AM] [Dima Style] didFinishLaunching. 
[6/10/2017, 11:52:39 AM] Homebridge is running on port 51826.
[6/10/2017, 11:52:39 AM] [Dima Style] Loading accessories 
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Диспенсер
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Ап.1 Движение
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Ап.1 Температура
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Ап. 1 Осветеност
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Ап.1 Бойлер Офиси
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Бойлер Офиси
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Площандка
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  да се ползва за Вент
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  UPS Стая Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  ВакуАктив
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Врата Стая Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Темп. Стая Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Кафе Машина
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Клима Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  Офис Влади
[6/10/2017, 11:52:39 AM] [Dima Style] Added/changed accessory:  UPS Eaton
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Дим Фризьoри 
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Темп.Въздух
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Реклама
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Педикюр
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  48V Захранване
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Сървър Вентилация 
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Фризьори
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  ДНК
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Аудио
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Рецепция Движение
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Рецепция
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Осветеност
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Врата
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  led рецепция
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Фризьори
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Влажност
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Външна Температура
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Влажност
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Подово Рецепция
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Темп. Рецепция
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Фризьори
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Подово
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Наводнение
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Темп. Наводнение
[6/10/2017, 11:52:40 AM] [Dima Style] Added/changed accessory:  Трева Напояване
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Капково Напояване
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Движение
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Бойлер
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Сървър Шкаф
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Темп. Движение
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Пералня
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Сушилня
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  UPS APC SC1000
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Бойлер
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Бойлер
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  2️⃣
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  3️⃣
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  1️⃣
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Вентилация
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  154.2
[6/10/2017, 11:52:41 AM] [Dima Style] Added/changed accessory:  Осв. между етажи
[6/10/2017, 11:52:41 AM] [Dima Style] Error getting data from Home Center:  TypeError: Cannot read property 'getCharacteristic' of undefined
    at Accessory._sideloadServices (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/Accessory.js:880:5)
    at PlatformAccessory._prepareAssociatedHAPAccessory (/usr/lib/node_modules/homebridge/lib/platformAccessory.js:148:32)
    at Server._handleRegisterPlatformAccessories (/usr/lib/node_modules/homebridge/lib/server.js:431:15)
    at Server.<anonymous> (/usr/lib/node_modules/homebridge/lib/server.js:36:10)
    at emitOne (events.js:96:13)
    at API.emit (events.js:191:7)
    at API.registerPlatformAccessories (/usr/lib/node_modules/homebridge/lib/api.js:159:8)
    at ShadowAccessory.resgisterUpdateccessory (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/shadows.js:88:17)
    at FibaroHC2.addAccessory (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/index.js:158:25)
    at LoadAccessories.devices.map (/usr/lib/node_modules/homebridge-fibaro-hc2/dist/index.js:105:22)
[6/10/2017, 11:52:49 AM] Mute : [false]
[6/10/2017, 11:52:49 AM] Busy with alarm now - not getting status
[6/10/2017, 11:52:49 AM] Got status
[6/10/2017, 11:52:49 AM] Results:
[6/10/2017, 11:52:49 AM] Zone 4 off (Етаж 2 Ап.1)
[6/10/2017, 11:52:49 AM] Zone 19 off (Етаж 2 ВИП)
[6/10/2017, 11:52:49 AM] Zone 20 off (Офис Влади)
[6/10/2017, 11:52:49 AM] Zone 22 off (Етаж 2 Ап.2)
[6/10/2017, 11:52:49 AM] Alarmstatus :Disarmed
[6/10/2017, 11:52:59 AM] Mute : [false]
[6/10/2017, 11:52:59 AM] Busy with alarm now - not getting status
[6/10/2017, 11:52:59 AM] Got status
[6/10/2017, 11:52:59 AM] Results:
[6/10/2017, 11:52:59 AM] Zone 4 off (Етаж 2 Ап.1)
[6/10/2017, 11:52:59 AM] Zone 19 off (Етаж 2 ВИП)
[6/10/2017, 11:52:59 AM] Zone 20 off (Офис Влади)
[6/10/2017, 11:52:59 AM] Zone 22 off (Етаж 2 Ап.2)
[6/10/2017, 11:52:59 AM] Alarmstatus :Disarmed
[6/10/2017, 11:53:09 AM] Mute : [false]
[6/10/2017, 11:53:09 AM] Busy with alarm now - not getting status
[6/10/2017, 11:53:09 AM] Got status
[6/10/2017, 11:53:09 AM] Results:
[6/10/2017, 11:53:09 AM] Zone 4 off (Етаж 2 Ап.1)
[6/10/2017, 11:53:09 AM] Zone 19 off (Етаж 2 ВИП)
[6/10/2017, 11:53:09 AM] Zone 20 off (Офис Влади)
[6/10/2017, 11:53:09 AM] Zone 22 off (Етаж 2 Ап.2)
[6/10/2017, 11:53:09 AM] Alarmstatus :Disarmed
[6/10/2017, 11:53:19 AM] Mute : [false]
[6/10/2017, 11:53:19 AM] Busy with alarm now - not getting status
[6/10/2017, 11:53:19 AM] Got status
[6/10/2017, 11:53:19 AM] Results:
[6/10/2017, 11:53:19 AM] Zone 4 off (Етаж 2 Ап.1)
[6/10/2017, 11:53:19 AM] Zone 19 off (Етаж 2 ВИП)
[6/10/2017, 11:53:19 AM] Zone 20 off (Офис Влади)
[6/10/2017, 11:53:19 AM] Zone 22 off (Етаж 2 Ап.2)
[6/10/2017, 11:53:19 AM] Alarmstatus :Disarmed

**Comment by agisofttm** [comment]
@ilcato - the issue is resolved - the problem is this smart meter "Netichome DIN Smart Meter", also called Quibino Smart meter. Because he doesn't have a template, I needed to configure it from parameters, but on fibaro UI it shows "Unconfigured", so I removed it and everything is perfect again.... is there a way to ignore Unconfigured devices in your plugin?

**Comment by ilcato** [comment]
Name it with an _ in front

**Comment by agisofttm** [comment]
@ilcato - Thank you very much for the help. Sorry again

**Comment by Mx-Razvan** [comment]
Hi Ilcato

after the update to 2.0.4 the AC Control accessory is blocked on hating - 21 degrees. It shows that is on all the time. The temperature setup is working but I cannot put it on cool or off (like before, unless it didn't work in HC2)

also - on the global variables - made according to the wiki - just the first one from the list is working - switching from true to false and back in HC2. The second one and the rest aren't working.

Any ideas why?

Thanks

**Comment by ilcato** [comment]
which was the previous version of your setup? 2.0.3 or before?

**Comment by Mx-Razvan** [comment]
1.1.2

**Comment by ilcato** [comment]
In 1.1.2 you were not able to change the cooling state, I mean you were able to change it from the Home app but the state was not propagated to the Home Center and viceversa.

**Comment by Mx-Razvan** [comment]
Yes - I was able to change the state and it remained as I set it in home app, without being propagated to HC2. Just the temperature set worked. Now the temperature set is working and propagated to HC2 but when I try to put it on cool, off or auto in Home app it keeps returning to Heat 21 degrees

**Comment by ilcato** [comment]
ok, just publish 2.0.5 that by default behave like 1.1.2.
The new logic is for heating thermostats like Danfoss in order to manage the cooling state that is not present in the z-wave device.

**Comment by ilcato** [comment]
For other device I still need to implement the logic for managing the cooling state.
I do not have this kind of devices so it is not a priority for me.

**Comment by Mx-Razvan** [comment]
Understood. I'll update and see how is working

**Comment by gondiciembre** [comment]
Hi there ilcato ! sorry to bother, do you still plan to support  Remotec ? thanks for all your work!

**Comment by ilcato** [comment]
Not without a real device to test

**Comment by Mx-Razvan** [comment]
Hi ilcato!
As i have uninstalled the HCL and the accesories from another location I’m able to send you the spare remotec controler for test if you want. Just let me know

**Comment by ilcato** [comment]
@Mx-Razvan, sure.

**Comment by Mx-Razvan** [comment]
Ok. Send me the shipping details at mx.razvan@gmail.com

**Comment by ilcato** [comment]
@Auto-m8r, @gondiciembre, @agisofttm, thanks to @Mx-Razvan that provided me a Remotec device I was able to implement a full Remotec zxt-120 support. I committed the code on GitHub but not published on npm. Please try the new functionality by updating the "dest" directory on your setup.
Thanks.

**Comment by agisofttm** [comment]
@ilcato - Thats a great news - I'll try it as soon as possible 😁

**Comment by gondiciembre** [comment]
@ilcato I have just tested it and it is working great ! thanks so much for working on this.
